libs/sql-parser/src/main/java/io/crate/sql/Identifiers.java

```diff
@@ -148,7 +142,11 @@ private static boolean isLowerCase(char c) {
 
     private static boolean reserved(String expression) {
         try {
-            return !(SqlParser.createExpression(expression) instanceof QualifiedNameReference);
+            Expression expr = SqlParser.createExpression(expression);
+            if (RESERVED_FUNCTIONS.contains(expression.toLowerCase(Locale.ENGLISH))) {
+                return true;
+            }
+            return !(expr instanceof QualifiedNameReference || expr instanceof FunctionCall);
```

Comment:
Is the `|| expr instanceof FunctionCall` needed? Looks like only the listed function are reserved which returns true then already...